infrastructure/scripts/ensure_venv.py

```
@@ -0,0 +1,93 @@
+    if not venv_ver or sys_ver != venv_ver:
+        print(f"[ensure_venv] Recreating .venv (sys={sys_ver}, venv={venv_ver})...")
+        if os.path.exists(venv_dir):
+            shutil.rmtree(venv_dir)
+
+        subprocess.run(
+            [system_python, "-m", "venv", "--without-pip", venv_dir], check=True
+        )
+
+        # Bootstrap pip
+        get_pip_url = "https://bootstrap.pypa.io/get-pip.py"
+        temp_dir = os.environ.get("TEMP", os.environ.get("TMP", "/tmp"))
```

Ruff (0.15.12)

[error] Probable insecure usage of temporary file or directory: "/tmp"

(S108)